scripts/testgen/TestGen.hx

```diff
@@ -0,0 +1,342 @@
+import haxe.Template;
+import haxe.Json;
+import sys.io.File;
+import sys.FileSystem;
+import sys.Http;
+
+using StringTools;
+
+class TestGen {
+    static var exercise: String; 
+    static var exerciseStubTmpl = new Template(
+"package;
+
+class ::exercise:: {
+    ::methods:: 
+}"
+    );
+
+    static var methodsTmpl = new Template(
+    "public static function ::method::(::args::): ::returnType:: {
+        throw \"Not Implemented\"; // Delete this statement and write your own implementation.
+    }"
+    );
+
+    static var testTmpl = new Template(
+"package;
+
+using buddy.Should;
+
+//Created by testgen.hx
+class Test extends buddy.SingleSuite {
+    public function new() {
+        ::stories::
+    }
+}"
+    );
+
+    static var storyTmpl = new Template(
+        "describe(\"::story::\", {
+        ::testCases::});"
+    );
+
+    static var testCaseTmpl = new Template(
+        "    it(\"::testCase::\", {
+                ::testCriteria::
+            });
+        "
+    );
+
+    public static function main() {
+        var dataUrl            = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/canonical-data.json");
+        var descriptionUrl     = new Template("https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/::exercise::/description.md");
```

Comment:
`./bin/configlet sync --docs --filepaths --metadata --tests include -uye <slug>` would create the folders and files we need. We'd still need to fetch the canonical data json but we might be able to get it from configlet's cached copy of the problem-specifications repo.